torchsynth/synth.py

```diff
@@ -158,12 +166,28 @@ def _forward(self) -> T:
         # The convention for triggering a note event is that it has
         # the same note_on_duration for both ADSRs.
         midi_f0, note_on_duration = self.keyboard()
-        pitch_envelope = self.pitch_adsr.forward(note_on_duration)
-        amp_envelope = self.amp_adsr.forward(note_on_duration)
 
-        audio_out = self.vco_1.forward(midi_f0, pitch_envelope)
-        vco_2_out = self.vco_2.forward(midi_f0, pitch_envelope)
+        # Compute LFOs with envelopes
+        lfo_1 = self.lfo_1() * self.lfo_1_adsr(note_on_duration)
+        lfo_2 = self.lfo_2() * self.lfo_2_adsr(note_on_duration)
```

Comment:
Open to different setups on this. two LFOs with different envelopes applied to them leads to some cool movement between the two modulation signals. And then these are mixed at various levels with the pitch and amplitude envelope.